internal/api/grpc/admin/sms_converter.go

```diff
@@ -64,17 +81,39 @@ func smsStateToPb(state domain.SMSConfigState) settings_pb.SMSProviderConfigStat
 	}
 }
 
-func AddSMSConfigTwilioToConfig(req *admin_pb.AddSMSProviderTwilioRequest) *twilio.Config {
-	return &twilio.Config{
-		SID:          req.Sid,
-		SenderNumber: req.SenderNumber,
-		Token:        req.Token,
+func addSMSConfigTwilioToConfig(ctx context.Context, req *admin_pb.AddSMSProviderTwilioRequest) *command.AddTwilioConfig {
+	return &command.AddTwilioConfig{
+		ResourceOwner: authz.GetInstance(ctx).InstanceID(),
+		Description:   req.Description,
+		SID:           req.Sid,
+		SenderNumber:  req.SenderNumber,
+		Token:         req.Token,
+	}
+}
+
+func updateSMSConfigTwilioToConfig(ctx context.Context, req *admin_pb.UpdateSMSProviderTwilioRequest) *command.ChangeTwilioConfig {
+	return &command.ChangeTwilioConfig{
+		ResourceOwner: authz.GetInstance(ctx).InstanceID(),
+		ID:            req.Id,
+		Description:   gu.Ptr(req.Description),
+		SID:           gu.Ptr(req.Sid),
+		SenderNumber:  gu.Ptr(req.SenderNumber),
```

Comment:
why do we need them as pointer?

Reply:
That's from the partial update, as I mentioned command-side could handle it, that's why I left also the pointers.